game/src/ecs/components/HealthComponent.java

```diff
@@ -0,0 +1,178 @@
+     * Creates a HealthComponent with default values
+     *
+     * @param entity associated entity
+     */
+    public HealthComponent(@DSLContextMember(name = "entity") Entity entity) {
+        super(entity);
+        this.maximalHitPoints = 1;
+        this.currentHitPoints = 1;
+        this.onDeath = entity2 -> {};
+        this.getHitAnimation = new Animation(missingTexture, 100);
+        this.dieAnimation = new Animation(missingTexture, 100);
+        damageToGet = new ArrayList<>();
+    }
+
+    /**
+     * Adds damage, which is accounted for by the system
+     *
+     * @param damage Damage that should be inflicted
+     */
+    public void getHit(Damage damage) {
+        damageToGet.add(damage);
+    }
+
+    /** Triggers the onDeath Function */
+    public void triggerOnDeath() {
+        onDeath.onDeath(entity);
+    }
+
+    /**
+     * @return List with all damage objects that still need to be accounted for
+     */
+    public List<Damage> getDamageList() {
+        return damageToGet;
+    }
+
+    /**
+     * Calculate the amount of damage of a certain type
+     *
+     * @param dt Type of damage object that still need to be accounted for
+     * @return Sum of all damage objects of type dt (default: 0)
+     */
+    public int getDamage(DamageType dt) {
+        return damageToGet.stream()
+                .filter(d -> d.damageType() == dt)
+                .mapToInt(Damage::damageAmount)
+                .sum();
+    }
+
+    /** Clear the damage list */
+    public void clearDamageList() {
+        damageToGet.clear();
+    }
+
+    /**
+     * Sets the current life points, capped at the value of the maximum hit-points
+     *
+     * @param amount new amount of current hit-points
+     */
+    public void setCurrentHitPoints(int amount) {
+        this.currentHitPoints = Math.min(maximalHitPoints, amount);
+    }
+
+    /**
+     * Sets the value of the Maximum Hit-points. If the new maximum hit-points are less than the
+     * current hit-points, the current points are set to the new maximum hit-points.
+     *
+     * @param amount new amount of maximal hit-points
+     */
+    public void setMaximalHitPoints(int amount) {
+        this.maximalHitPoints = amount;
+        currentHitPoints = Math.min(currentHitPoints, maximalHitPoints);
+    }
+
+    /**
+     * Set the animation to be played when the entity dies
+     *
+     * @param dieAnimation new dieAnimation
+     */
+    public void setDieAnimation(Animation dieAnimation) {
+        this.dieAnimation = dieAnimation;
+    }
+
+    /**
+     * Set the animation to be played when the entity is hit
+     *
+     * @param isHitAnimation new isHitAnimation
+     */
+    public void setGetHitAnimation(Animation isHitAnimation) {
+        this.getHitAnimation = isHitAnimation;
+    }
+
+    /**
+     * Set a new function to be called when dying.
+     *
+     * @param onDeath new onDeath function
+     */
+    public void setOnDeath(IOnDeathFunction onDeath) {
```

Comment:
Diese Methode ist konzeptionell fragwürdig: Sollte der Callback nicht beim Erstellen der Component fest sein (und bleiben)? Gibt es einen Use-Case für das Ändern des Callbacks in der HealthComponent zu einem späteren Zeitpunkt? Im Code habe ich nur einen Test gefunden, sonst wird die Methode nicht verwendet.

Reply:
Hier wäre denkbar, dass ein gesundes Item irgendwas beim sterben macht (mich retten). Oder als Bestrafung für eine falsche Antwort werden alle Monster verflucht und explodieren beim tot nochmal.

Reply:
Hmmm, verstehe. Irgendwie tue ich mich schwer mit dem Zuschnitt dieser Component.

Wäre es mit dem "gefundenes Item rettet mich" nicht eher so, dass mich das HealthSystem dann rettet (nicht die Component) und mir wieder positive Gesundheit und keinen Damage oder sowas zuschreibt, d.h. ich sterbe in dem Fall gar nicht erst? 

Und hat "ich bin verflucht" was mit meiner Gesundheit zu tun? Oder wäre das nicht eher eine verwandte Component, die dann beim tatsächlichen Sterben wg. mangelnder Health die Animation in der AnimationsComponent überschreibt (über das jeweilige System)?

Reply:
> Gesundheit und Tot gehören zusammen.

Das habe ich auch nicht verneint.

> 
> > Wäre es mit dem "gefundenes Item rettet mich" nicht eher so, dass mich das HealthSystem dann rettet (nicht die Component) und mir wieder positive Gesundheit und keinen Damage oder sowas zuschreibt, d.h. ich sterbe in dem Fall gar nicht erst?
> 
> Da könnten wir uns jetzt über Implementierungsdetails kloppen. Ich würde das Item so implementieren, dass es meine `onDeath` Methode überschreibt und mir wieder x-Lebenspunkte gibt, wenn ich gestorben bin.

Das ist eher ein konzeptionelles Detail. Wenn mich ein Item rettet, sollte mir das zuständige System meine Gesundheitspunkte wieder resetten. Das hat für mich nichts mit "onDeath" zu tun.

Reply:
> Das ist eher ein konzeptionelles Detail. Wenn mich ein Item rettet, sollte mir das zuständige System meine Gesundheitspunkte wieder resetten. Das hat für mich nichts mit "onDeath" zu tun.

Ist halt die Frage was das Item macht. Wenn es ein Heiltrank ist, dann ja. Aber vielleicht ist es ja eine Fee (okay, das hab ich von Zelda geklaut), die erst aktiviert wird, wenn du tot bist. In Zelda stirbst du (mit Animation, tragischer Musik etc) und dann kommt die Fee aus deinen Inventar und belebt dich wieder.

Reply:
Vielleicht ist das Beispiel item welches mich rettet wirklich dafür ungeeignet.  z.b. Bei Tod des Helden starte Spiel neu oder bei Tod des Monsters lasse Gegenstand in die Welt fallen. Klingen für mich plausible Logiken, welche beim Tod passieren und welche ggfls. Frei konfiguriert werden sollen. Vielleicht einfach einmal in der Planungsrunde besprechen.

Reply:
> Vielleicht einfach einmal in der Planungsrunde besprechen.

Find ich gut. Dann verschieben wir das hier nach #353